src/view.jl

```diff
@@ -46,9 +62,17 @@ V = rand(500);
 # was 0.140318 seconds seconds, now 0.030957 seconds, factor 4
 =#
 
-# performance: avoid reshaping lazy transposes, as this is very slow in broadcasting
+# performance: avoid reshaping lazy transposes (apart from GPU arrays), as this is very slow in broadcasting
 const LazyPerm = Union{PermutedDimsArray, LinearAlgebra.Transpose, LinearAlgebra.Adjoint}
-orient(A::Union{LazyPerm, SubArray{<:Any,<:Any,<:LazyPerm}}, code::Tuple) = orient(collect(A), code)
+orient(A::Union{LazyPerm, SubArray{<:Any,<:Any,<:LazyPerm}}, code::Tuple) = begin
+    if storage_type(A) <: GPUArray
+        # call the original algorithm with reshape, since collect would copy to CPU
+        # fortunately, this is not slow on GPU
+        invoke(orient,Tuple{AbstractArray,Tuple},A,code)
```

Comment:
```suggestion
        _orient(A,code)
```
Now there's a function just for the reshaping.